[AUTOCUT] Gradle Check Flaky Test Report for RemoteClusterStateServiceTests

## Flaky Test Report for `RemoteClusterStateServiceTests`

Noticed the `RemoteClusterStateServiceTests` has some flaky, failing tests that failed during **post-merge actions** and **timer trigerred run on main branch**.

### Details

| Git Reference | Merged Pull Request | Build Details | Test Name |
|---------------|----------------------|---------------|-----------|
| main | [null](https://github.com/opensearch-project/OpenSearch/pull/null) | [63793](https://build.ci.opensearch.org/job/gradle-check/63793/testReport/) <br><br>[63844](https://build.ci.opensearch.org/job/gradle-check/63844/testReport/) <br><br>[63860](https://build.ci.opensearch.org/job/gradle-check/63860/testReport/) <br><br>[63888](https://build.ci.opensearch.org/job/gradle-check/63888/testReport/) <br><br>[63907](https://build.ci.opensearch.org/job/gradle-check/63907/testReport/) <br><br>[63916](https://build.ci.opensearch.org/job/gradle-check/63916/testReport/) <br><br>[63972](https://build.ci.opensearch.org/job/gradle-check/63972/testReport/) <br><br>[63984](https://build.ci.opensearch.org/job/gradle-check/63984/testReport/) <br><br>[63985](https://build.ci.opensearch.org/job/gradle-check/63985/testReport/) <br><br>[63988](https://build.ci.opensearch.org/job/gradle-check/63988/testReport/) <br><br>[64007](https://build.ci.opensearch.org/job/gradle-check/64007/testReport/) <br><br>[64028](https://build.ci.opensearch.org/job/gradle-check/64028/testReport/) <br><br>[64044](https://build.ci.opensearch.org/job/gradle-check/64044/testReport/) <br><br>[64048](https://build.ci.opensearch.org/job/gradle-check/64048/testReport/) <br><br>[64101](https://build.ci.opensearch.org/job/gradle-check/64101/testReport/) <br><br>[64110](https://build.ci.opensearch.org/job/gradle-check/64110/testReport/) <br><br>[64138](https://build.ci.opensearch.org/job/gradle-check/64138/testReport/) <br><br>[64171](https://build.ci.opensearch.org/job/gradle-check/64171/testReport/) <br><br>[64227](https://build.ci.opensearch.org/job/gradle-check/64227/testReport/) <br><br>[64336](https://build.ci.opensearch.org/job/gradle-check/64336/testReport/) <br><br>[64357](https://build.ci.opensearch.org/job/gradle-check/64357/testReport/) <br><br>[64376](https://build.ci.opensearch.org/job/gradle-check/64376/testReport/) <br><br>[64397](https://build.ci.opensearch.org/job/gradle-check/64397/testReport/) <br><br>[64413](https://build.ci.opensearch.org/job/gradle-check/64413/testReport/) | `org.opensearch.gateway.remote.RemoteClusterStateServiceTests.testReadClusterStateInParallel_ExceptionDuringRead` |
| ad4c7f72888d8e754560402441805a7ba0d8394b | [19407](https://github.com/opensearch-project/OpenSearch/pull/19407) | [64390](https://build.ci.opensearch.org/job/gradle-check/64390/testReport/) | `org.opensearch.gateway.remote.RemoteClusterStateServiceTests.testReadClusterStateInParallel_ExceptionDuringRead` |
| 488149fd3a02af454019411e71b9f29c9ef125c9 | [19220](https://github.com/opensearch-project/OpenSearch/pull/19220) | [64182](https://build.ci.opensearch.org/job/gradle-check/64182/testReport/) | `org.opensearch.gateway.remote.RemoteClusterStateServiceTests.testReadClusterStateInParallel_ExceptionDuringRead` |
| 28597dcc69b03b4e12f805f04f289c2545c1dd60 | [19191](https://github.com/opensearch-project/OpenSearch/pull/19191) | [63874](https://build.ci.opensearch.org/job/gradle-check/63874/testReport/) | `org.opensearch.gateway.remote.RemoteClusterStateServiceTests.testReadClusterStateInParallel_ExceptionDuringRead` |
| 5babdca18de4a20b4674bd056ec9da196b6cbe68 | [19280](https://github.com/opensearch-project/OpenSearch/pull/19280) | [63821](https://build.ci.opensearch.org/job/gradle-check/63821/testReport/) | `org.opensearch.gateway.remote.RemoteClusterStateServiceTests.testReadClusterStateInParallel_ExceptionDuringRead` |
| e82cd9e22cb04bf6ac1481845e50dda7272fc52a | [19088](https://github.com/opensearch-project/OpenSearch/pull/19088) | [63803](https://build.ci.opensearch.org/job/gradle-check/63803/testReport/) | `org.opensearch.gateway.remote.RemoteClusterStateServiceTests.testReadClusterStateInParallel_ExceptionDuringRead` |

The other pull requests, besides those involved in post-merge actions, that contain failing tests with the `RemoteClusterStateServiceTests` class are:

- [13924](https://github.com/opensearch-project/OpenSearch/pull/13924)
- [16271](https://github.com/opensearch-project/OpenSearch/pull/16271)
- [15478](https://github.com/opensearch-project/OpenSearch/pull/15478)
- [15739](https://github.com/opensearch-project/OpenSearch/pull/15739)
- [15740](https://github.com/opensearch-project/OpenSearch/pull/15740)
- [15382](https://github.com/opensearch-project/OpenSearch/pull/15382)
- [14006](https://github.com/opensearch-project/OpenSearch/pull/14006)
- [14338](https://github.com/opensearch-project/OpenSearch/pull/14338)
- [14668](https://github.com/opensearch-project/OpenSearch/pull/14668)
- [15100](https://github.com/opensearch-project/OpenSearch/pull/15100)
- [14677](https://github.com/opensearch-project/OpenSearch/pull/14677)
- [14671](https://github.com/opensearch-project/OpenSearch/pull/14671)
- [19142](https://github.com/opensearch-project/OpenSearch/pull/19142)
- [13577](https://github.com/opensearch-project/OpenSearch/pull/13577)
- [15937](https://github.com/opensearch-project/OpenSearch/pull/15937)
- [15145](https://github.com/opensearch-project/OpenSearch/pull/15145)
- [19008](https://github.com/opensearch-project/OpenSearch/pull/19008)
- [19174](https://github.com/opensearch-project/OpenSearch/pull/19174)
- [19332](https://github.com/opensearch-project/OpenSearch/pull/19332)
- [19358](https://github.com/opensearch-project/OpenSearch/pull/19358)
- [15218](https://github.com/opensearch-project/OpenSearch/pull/15218)
- [18550](https://github.com/opensearch-project/OpenSearch/pull/18550)
- [19181](https://github.com/opensearch-project/OpenSearch/pull/19181)
- [19306](https://github.com/opensearch-project/OpenSearch/pull/19306)
- [19320](https://github.com/opensearch-project/OpenSearch/pull/19320)
- [19344](https://github.com/opensearch-project/OpenSearch/pull/19344)
- [19356](https://github.com/opensearch-project/OpenSearch/pull/19356)
- [19373](https://github.com/opensearch-project/OpenSearch/pull/19373)
- [15424](https://github.com/opensearch-project/OpenSearch/pull/15424)
- [15929](https://github.com/opensearch-project/OpenSearch/pull/15929)
- [15978](https://github.com/opensearch-project/OpenSearch/pull/15978)
- [18019](https://github.com/opensearch-project/OpenSearch/pull/18019)
- [19005](https://github.com/opensearch-project/OpenSearch/pull/19005)
- [19012](https://github.com/opensearch-project/OpenSearch/pull/19012)
- [19116](https://github.com/opensearch-project/OpenSearch/pull/19116)
- [19152](https://github.com/opensearch-project/OpenSearch/pull/19152)
- [19178](https://github.com/opensearch-project/OpenSearch/pull/19178)
- [19184](https://github.com/opensearch-project/OpenSearch/pull/19184)
- [19209](https://github.com/opensearch-project/OpenSearch/pull/19209)
- [19240](https://github.com/opensearch-project/OpenSearch/pull/19240)
- [19263](https://github.com/opensearch-project/OpenSearch/pull/19263)
- [19278](https://github.com/opensearch-project/OpenSearch/pull/19278)
- [19309](https://github.com/opensearch-project/OpenSearch/pull/19309)
- [19316](https://github.com/opensearch-project/OpenSearch/pull/19316)
- [19340](https://github.com/opensearch-project/OpenSearch/pull/19340)
- [19347](https://github.com/opensearch-project/OpenSearch/pull/19347)
- [19348](https://github.com/opensearch-project/OpenSearch/pull/19348)
- [19352](https://github.com/opensearch-project/OpenSearch/pull/19352)
- [19359](https://github.com/opensearch-project/OpenSearch/pull/19359)
- [19368](https://github.com/opensearch-project/OpenSearch/pull/19368)
- [19376](https://github.com/opensearch-project/OpenSearch/pull/19376)
- [19377](https://github.com/opensearch-project/OpenSearch/pull/19377)
- [19396](https://github.com/opensearch-project/OpenSearch/pull/19396)
- [19397](https://github.com/opensearch-project/OpenSearch/pull/19397)
- [19402](https://github.com/opensearch-project/OpenSearch/pull/19402)
- [19405](https://github.com/opensearch-project/OpenSearch/pull/19405)
- [19413](https://github.com/opensearch-project/OpenSearch/pull/19413)
- [19416](https://github.com/opensearch-project/OpenSearch/pull/19416)
- [19425](https://github.com/opensearch-project/OpenSearch/pull/19425)
- [19432](https://github.com/opensearch-project/OpenSearch/pull/19432)

For more details on the failed tests refer to [OpenSearch Gradle Check Metrics](https://metrics.opensearch.org/_dashboards/app/dashboards#/view/e5e64d40-ed31-11ee-be99-69d1dbc75083) dashboard.

```
REPRODUCE WITH: ./gradlew ':server:test' --tests 'org.opensearch.gateway.remote.RemoteClusterStateServiceTests.testReadClusterStateInParallel_ExceptionDuringRead' -Dtests.seed=7D187563E049ADC9 -Dtests.security.manager=true -Dtests.jvm.argline="-XX:TieredStopAtLevel=1 -XX:ReservedCodeCacheSize=64m" -Dtests.locale=saq-Latn-KE -Dtests.timezone=Pacific/Tahiti -Druntime.java=24

RemoteClusterStateServiceTests > testReadClusterStateInParallel_ExceptionDuringRead FAILED
    java.lang.AssertionError: 
    Expected: (an empty collection or iterable containing [a string starting with "No Root logger was configured, creating default ERROR-level Root logger with Console appender"] or iterable containing [a string starting with "No Root logger was configured, creating default ERROR-level Root logger with Console appender", a string starting with "No Root logger was configured, creating default ERROR-level Root logger with Console appender"])
         but: was <[An exception occurred processing Appender console, An exception occurred processing Appender console, An exception occurred processing Appender console]>
        at __randomizedtesting.SeedInfo.seed([7D187563E049ADC9:837EBCCE32AFF7B9]:0)
        at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
        at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:6)
        at org.apache.lucene.tests.util.LuceneTestCase.assertThat(LuceneTestCase.java:2104)
        at org.opensearch.test.OpenSearchTestCase.checkStaticState(OpenSearchTestCase.java:740)
        at org.opensearch.test.OpenSearchTestCase.after(OpenSearchTestCase.java:506)
        at java.****/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
        at java.****/java.lang.reflect.Method.invoke(Method.java:565)
        at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1750)
        at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:996)
        at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
        at org.junit.rules.RunRules.evaluate(RunRules.java:20)
        at org.apache.lucene.tests.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:48)
        at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
        at org.apache.lucene.tests.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:45)
        at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
        at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
        at org.junit.rules.RunRules.evaluate(RunRules.java:20)
        at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
        at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:368)
        at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:817)
        at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:468)
        at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:947)
        at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:832)
        at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:883)
        at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:894)
        at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
        at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
        at org.apache.lucene.tests.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:38)
        at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
        at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
        at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
        at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
        at org.apache.lucene.tests.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
        at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
        at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
        at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
        at org.apache.lucene.tests.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:47)
        at org.junit.rules.RunRules.evaluate(RunRules.java:20)
        at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
        at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:368)
        at java.****/java.lang.Thread.run(Thread.java:1447)
```

--- a/server/src/test/java/org/opensearch/gateway/remote/RemoteClusterStateServiceTests.java
+++ b/server/src/test/java/org/opensearch/gateway/remote/RemoteClusterStateServiceTests.java
@@ -1344,8 +1344,8 @@ public class RemoteClusterStateServiceTests extends OpenSearchTestCase {
         ClusterState previousClusterState = generateClusterStateWithAllAttributes().build();
         ClusterMetadataManifest manifest = generateClusterMetadataManifestWithAllAttributes().build();
         BlobContainer container = mockBlobStoreObjects();
-        Exception mockException = new IOException("mock exception");
-        when(container.readBlob(anyString())).thenThrow(mockException);
+        String exceptionMsg = "mock exception";
+        when(container.readBlob(anyString())).thenAnswer(inv -> { throw new IOException(exceptionMsg); });
         remoteClusterStateService.start();
         RemoteStateTransferException exception = expectThrows(
             RemoteStateTransferException.class,
@@ -1371,7 +1371,7 @@ public class RemoteClusterStateServiceTests extends OpenSearchTestCase {
         );
         assertEquals("Exception during reading cluster state from remote", exception.getMessage());
         assertTrue(exception.getSuppressed().length > 0);
-        assertEquals(mockException, exception.getSuppressed()[0].getCause());
+        assertEquals(exceptionMsg, exception.getSuppressed()[0].getCause().getMessage());
     }
 
     public void testReadClusterStateInParallel_UnexpectedResult() throws IOException {

